appinventor/appengine/src/com/google/appinventor/client/wizards/FileUploadWizard.java

```diff
@@ -218,4 +240,58 @@ private void finishUpload(FolderNode folderNode, String filename,
       fileUploadedCallback.onFileUploaded(folderNode, uploadedFileNode);
     }
   }
+
+  private void createErrorDialog(String errorType, final FolderNode folderNode,
+      final FileUploadedCallback fileUploadedCallback) {
+    final DialogBox dialogBox = new DialogBox(false,true);
+    HTML message;
+    dialogBox.setStylePrimaryName("ode-DialogBox");
+    dialogBox.setHeight("150px");
+    dialogBox.setWidth("350px");
+    dialogBox.setGlassEnabled(true);
+    dialogBox.setAnimationEnabled(true);
+    dialogBox.center();
+    VerticalPanel DialogBoxContents = new VerticalPanel();
+    FlowPanel holder = new FlowPanel();
+    Button ok = new Button ("OK");
+    ok.addClickListener(new ClickListener() {
+      public void onClick(Widget sender) {
+        dialogBox.hide();
+        new FileUploadWizard(folderNode, fileUploadedCallback).show();
+      }
+    });
+    holder.add(ok);
+    if (errorType.equals("aia")) {
+      dialogBox.setText("Error: Cannot upload .aia file as media asset");  
+      message = new HTML(MESSAGES.aiaMediaAsset());
+      
+      Button info = new Button ("More Info");
+      info.addClickListener(new ClickListener() {
+        public void onClick(Widget sender) {
+          Window.open("http://appinventor.mit.edu/explore/ai2/share.html", 
+            "AIA_Help", "");
+        }
+      });
+      holder.add(info);
+    } else if (errorType.equals("noFile")) {
```

Comment:
You should get rid of this bunch of if ... then else if ... stuff. Instead just pass the error message to the dialog. as in:

`createErrorDialog(MESSAGES.noFileSelected(), folderNode, fileUploadedCallback);`

-Jeff